controller/util/cf_v2_client.go

```diff
@@ -0,0 +1,172 @@
+	err = ResponseBodyToObject(resp, &catalog)
+	if err != nil {
+		log.Printf("Failed to unmarshal catalog: %#v", err)
+		return nil, err
+	}
+
+	return &catalog, nil
+}
+
+func (c *cfV2BrokerClient) CreateServiceInstance(ID string, req *model.ServiceInstanceRequest) (*model.ServiceInstance, error) {
+	jsonBytes, err := json.Marshal(req)
+	if err != nil {
+		return nil, err
+	}
+
+	url := fmt.Sprintf(serviceInstanceFormatString, c.broker.BrokerURL, ID)
+
+	// TODO: Handle the auth
+	createHTTPReq, err := http.NewRequest("PUT", url, bytes.NewReader(jsonBytes))
+	client := &http.Client{}
+	log.Printf("Doing a request to: %s", url)
+	resp, err := client.Do(createHTTPReq)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	// TODO: Align this with the actual response model.
+	si := model.ServiceInstance{}
+	if err := ResponseBodyToObject(resp, &si); err != nil {
+		return nil, err
+	}
+	return &si, nil
+}
+
+func (c *cfV2BrokerClient) UpdateServiceInstance(ID string, req *model.ServiceInstanceRequest) (*model.ServiceInstance, error) {
+	return nil, fmt.Errorf("Not implemented")
+}
+
+func (c *cfV2BrokerClient) DeleteServiceInstance(ID string) error {
+	url := fmt.Sprintf(serviceInstanceFormatString, c.broker.BrokerURL, ID)
+
+	// TODO: Handle the auth
+	deleteHTTPReq, err := http.NewRequest("DELETE", url, nil)
+	if err != nil {
+		log.Printf("Failed to create new HTTP request: %v", err)
+		return err
+	}
+
+	client := &http.Client{}
+	log.Printf("Doing a request to: %s", url)
+	resp, err := client.Do(deleteHTTPReq)
+	if err != nil {
+		log.Printf("Failed to DELETE: %#v", err)
+		return err
+	}
+	defer resp.Body.Close()
+
+	return nil
+}
+
+func (c *cfV2BrokerClient) CreateServiceBinding(sID, bID string, req *model.BindingRequest) (*model.CreateServiceBindingResponse, error) {
+	jsonBytes, err := json.Marshal(req)
+	if err != nil {
+		log.Printf("Failed to marshal: %#v", err)
+		return nil, err
+	}
+
+	url := fmt.Sprintf(bindingFormatString, c.broker.BrokerURL, sID, bID)
+
+	// TODO: Handle the auth
+	createHTTPReq, err := http.NewRequest("PUT", url, bytes.NewReader(jsonBytes))
+	client := &http.Client{}
+	log.Printf("Doing a request to: %s", url)
+	resp, err := client.Do(createHTTPReq)
+	if err != nil {
+		log.Printf("Failed to PUT: %#v", err)
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	sbr := model.CreateServiceBindingResponse{}
+	err = ResponseBodyToObject(resp, &sbr)
+	if err != nil {
+		log.Printf("Failed to unmarshal: %#v", err)
+		return nil, err
+	}
+
+	return &sbr, nil
+}
+
+func (c *cfV2BrokerClient) DeleteServiceBinding(sID, bID string) error {
+	url := fmt.Sprintf(bindingFormatString, c.broker.BrokerURL, sID, bID)
+
+	// TODO: Handle the auth
+	deleteHTTPReq, err := http.NewRequest("DELETE", url, nil)
+	if err != nil {
+		log.Printf("Failed to create new HTTP request: %v", err)
+		return err
+	}
+
+	client := &http.Client{}
+	log.Printf("Doing a request to: %s", url)
+	resp, err := client.Do(deleteHTTPReq)
```

Comment:
Probably out of scope of this PR, but I'd like to introduce the idea here. I'd recommend using [ctxhttp](https://godoc.org/golang.org/x/net/context/ctxhttp) to 'harden' this and other HTTP requests to the external CF brokers. I've created https://github.com/kubernetes-incubator/service-catalog/issues/113 for this work. *Not blocking for this PR*